src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsDataRepairResource.kt

```diff
@@ -0,0 +1,40 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.adjudications
+
+import com.microsoft.applicationinsights.TelemetryClient
+import io.swagger.v3.oas.annotations.Operation
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class AdjudicationsDataRepairResource(
+  private val adjudicationService: AdjudicationsService,
+  private val telemetryClient: TelemetryClient,
+) {
+  @PostMapping("/prisons/{prisonId}/prisoners/{offenderNo}/adjudication/dps-charge-number/{chargeNumber}/punishments/repair")
```

Comment:
I did this for activities but without `/repair`. Just post for create and put for update.

Reply:
well this creates, deleted and updates all in one go